buffer_interface/buffer_interface.c

```diff
@@ -0,0 +1,90 @@
+#include <stdio.h>
+#include <stdint.h>
+
+#define BUFFER_SIZE 50
+
+uint32_t buffer1[BUFFER_SIZE] = {0};
+uint32_t buffer2[BUFFER_SIZE] = {0};
+
+enum buff_num {
+    BUFFER1,
+    BUFFER2
+};
+
+typedef enum {
+    false,
+    true
+} bool;
+
+bool check_unsigned_limit(uint8_t value, uint32_t limit)
+{
+    if(value >= limit || value < 0)
+    {
+        return false;
+    }
+    return true;
```

Comment:
It's a matter of taste, but a ternary operation might be a better fit here instead of if statement.

`return (value >= 0 && value < limit) ? true : false; `